src/Support/ResponseCache/ResponseCacheProfileByCountry.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ARKEcosystem\Foundation\Support\ResponseCache;
+
+use ARKEcosystem\Foundation\Support\Facades\Visitor;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Spatie\ResponseCache\CacheProfiles\CacheAllSuccessfulGetRequests;
+
+final class ResponseCacheProfileByCountry extends CacheAllSuccessfulGetRequests
+{
+    public function useCacheNameSuffix(Request $request): string
+    {
+        if (Auth::check()) {
+            return (string) Auth::id();
+        }
```

Comment:
do we need this as part of `ResponseCacheProfileByCountry`? 🤔 this is not related to country of origin and would also results in separate caches for each logged in user if registrations would be made open to the public at some point. Would be better as a separate class or to leave it for now until we run into a scenario where we need separate caches per user

Reply:
@ItsANameToo the default profile already separate by user (see the class that I'm extending) and the public site is different for logged users because they can see draft articles, I think that all the logged users see the exact same public site (since las time I checked draft articles are shown for All logged users not depending on the user/role for example)

So we can separate by loggenin+country but I think that using the id+country is better, is not like we are going to have hundred of users logged at the same time and that will make it this class more reusable on other projects 

What do you think?

Reply:
ah I see 🤔 fair enough for the auth thing then.